### PR TITLE
fix: allow un-revoking invitations

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/tasks/issues/666

We added some security restrictions around invitations recently (specifically, prevent sending more than 1 invite email per email per workspace per 24 hours).
But there are 2 problems:
- when you revoked an invitation that you sent in the last 24 hours, we block you from sending it again (this is good). But you don't have any way to un-revoke it. This change fixes this.

- when you add a long list of emails and some of them are for users that have already been invited in the last 24 hours, we fail the whole API call instead of at least sending those we can send.

### CODE REVIEW

File: front/pages/api/w/[wId]/invitations/index.ts [hunk 1]
```diff
@@ -154,41 +156,63 @@ async function handler(
...
           status_code: 400,
           api_error: {
-            type: "invalid_request_error",
-            message: `Some of the emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
+            type: "invitation_already_sent_recently",
+            message: `These emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
           },
         });
       }
+      await MembershipInvitation.update(
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/invitations/index.ts:202]: This should not live here. Please move to `front/lib/api`.

File: front/pages/w/[wId]/members/index.tsx [hunk 2]
```diff
@@ -1030,6 +1030,23 @@ async function sendInvitations({
...
+    }
+    if (data?.error?.type === "invitation_already_sent_recently") {
+      sendNotification({
+        type: "error",
+        title: emails.length === 1 ? "Invite failed" : "Invites failed",
+        description:
+          emails.length === 1
+            ? "This user has already been invited in the last 24 hours. Please wait before sending another invite."
+            : "These users have already been invited in the last 24 hours. Please wait before sending another invite.",
+      });
```
Comment (flvndvd) [inline front/pages/w/[wId]/members/index.tsx:1047]:
```suggestion
      sendNotification({
        type: "error",
        title: `Invite${emails.length > 1 ? "s" : ""} failed`,
        description: `${"emails.length === 1 ? "This user" : "These users"} have already been invited in the last 24 hours. Please wait before sending another invite.`,
      });
```

File: front/pages/api/w/[wId]/invitations/index.ts [hunk 3]
```diff
@@ -154,41 +156,63 @@ async function handler(
...
+        !unconsumedInvitations.revoked.length &&
+        invitationRequests.length > 0
       ) {
         return apiError(req, res, {
           status_code: 400,
           api_error: {
-            type: "invalid_request_error",
-            message: `Some of the emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
+            type: "invitation_already_sent_recently",
+            message: `These emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/pages/api/w/[wId]/invitations/index.ts:197]: Wait, no it's all indeed!

File: front/pages/api/w/[wId]/invitations/index.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/w/[wId]/invitations/index.ts]: Moved

File: front/pages/api/w/[wId]/invitations/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/api/w/[wId]/invitations/index.ts:197]: Yes the error is only if all invitations are converned

File: front/pages/w/[wId]/members/index.tsx (hunk 2 above)
Comment (fontanierh) [reply front/pages/w/[wId]/members/index.tsx:1047]: Meh, harder to read IMO. I'll meet you half way though and DRY the last part of the text that is indentical in both cases

**Review by PopDaph — APPROVED** [review]
🙏🏻

**Pull request merged**